### PR TITLE
Convert import export to new package config

### PR DESCRIPTION
**Pull request by leorossi**
## Description
Starting to convert `compass-import-export` package to Typescript

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [X] Misc

## Types of changes

- [ ] *Backport Needed*
- [X] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-import-export/src/utils/formatters.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ export const createJSONFormatter = function({ brackets = true } = {}) {
  */
 export const createCSVFormatter = function({ columns }) {
   return csv.format({
-    headers: columns,
+    headers: true,
```
Comment (mcasimir) [inline packages/compass-import-export/src/utils/formatters.js:49]: What is the reason for this change?

File: packages/compass-import-export/config/webpack.base.config.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ module.exports = {
       test: /\.node$/,
       use: 'node-loader'
     }, {
-      test: /node_modules[\\\/]JSONStream[\\\/]index\.js/,
+      test: /node_modules[\\/]JSONStream[\\/]index\.js/,
```
Comment (leorossi) [inline packages/compass-import-export/config/webpack.base.config.js:76]: @mcasimir is this ok? linter gave me error as un-necessary `\`

File: packages/compass-import-export/test/setup.js [hunk 3]
```diff
@@ -9,7 +9,9 @@ const chaiAsPromised = require('chai-as-promised');
 
 require('jsdom-global')();
 
+// eslint-disable-next-line no-undef
 window.Object = Object;
```
Comment (leorossi) [inline packages/compass-import-export/test/setup.js:13]: I could change eslint config to allow `window` as global, but I prefer to add annotations instead

File: packages/compass-import-export/src/utils/formatters.js (hunk 1 above)
Comment (leorossi) [reply packages/compass-import-export/src/utils/formatters.js]: it was a test, file was restored

File: packages/compass-import-export/package.json [hunk 4]
```diff
@@ -1,171 +1,78 @@
...
-    "pretest": "mongodb-runner start --port=27018",
-    "posttest": "mongodb-runner stop --port=27018",
-    "test:watch": "cross-env NODE_ENV=test mocha-webpack \"./src/**/*.spec.js\" --watch",
-    "test:dev": "cross-env NODE_ENV=test mocha-webpack",
-    "cover": "nyc npm run test",
-    "check": "npm run lint && npm run depcheck",
+    "bootstrap": "npm run compile",
     "prepublishOnly": "npm run compile",
-    "lint": "eslint \"./src/**/*.{js,jsx}\" \"./test/**/*.js\"",
+    "compile": "tsc -p tsconfig.json && gen-esm-wrapper . ./dist/.esm-wrapper.mjs",
```
Comment (gribnoysup) [inline packages/compass-import-export/package.json:33]: If we want to drop webpack build, then we need to remove the config files too (but this will also not work when trying to bundle less files), if we want to keep it and we are updating the plugin to new configuration, then we should use new shared webpack config (you can use create-workspace output for the plugin workspace as a reference for what configs and npm scripts should be part of the package)

File: packages/compass-import-export/test/setup.js [hunk 5]
```diff
@@ -9,7 +9,9 @@ const chaiAsPromised = require('chai-as-promised');
```
Comment (gribnoysup) [inline packages/compass-import-export/test/setup.js:9]: If you are moving to the new shared mocha config for tests in this package, you should not need this setup anymore as our shared configuration comes with a bunch of pre-configuration

File: packages/compass-import-export/package.json (hunk 4 above)
Comment (leorossi) [reply packages/compass-import-export/package.json:33]: I basically re-created the `compass-import-export` package with `create-workspace.js` script and then copied the files over and fixed eslint errors. Basically this PR will prepare the repo to use Typescript for the Exporter refactory

Comment (gribnoysup) [reply packages/compass-import-export/package.json:33]: When you were generating it, did you marked this package as plugin? Seems like some default plugin configs like the one for webpack are missing. Or are we intentionally not generating this package as plugin?

Comment (leorossi) [reply packages/compass-import-export/package.json:33]: I think I chose "no", I'll try regenerating it as plugin

File: packages/compass-import-export/config/webpack.base.config.js (hunk 2 above)
Comment (mcasimir) [reply packages/compass-import-export/config/webpack.base.config.js:76]: The "auto-fix" lamp in VSCode should do the right thing there. Though that's a bit strange, it doesn't complain in other packages.

File: packages/compass-import-export/package.json [hunk 6]
```diff
@@ -1,171 +1,90 @@
 {
   "name": "@mongodb-js/compass-import-export",
-  "productName": "Compass Import/Export",
-  "version": "6.21.0",
```
Comment (gribnoysup) [inline packages/compass-import-export/package.json:4]: This needs to be preserved

File: packages/compass-import-export/package.json [hunk 7]
```diff
@@ -1,171 +1,122 @@
...
+    "typecheck": "tsc --noEmit",
+    "eslint": "eslint",
+    "prettier": "prettier",
+    "lint": "npm run eslint . && npm run prettier -- --check .",
     "depcheck": "depcheck",
-    "test-ci": "npm run test",
-    "bootstrap": "npm run compile"
+    "check": "npm run typecheck && npm run lint && npm run depcheck",
+    "check-ci": "npm run check",
+    "pretest": "mongodb-runner start --port=27018",
```
Comment (gribnoysup) [inline packages/compass-import-export/package.json:46]: You need to ass the same pre-post script for test-electron command, because it's not there CI is failing currently

File: packages/compass-import-export/package.json [hunk 8]
```diff
@@ -1,171 +1,122 @@
...
-    "lint": "eslint \"./src/**/*.{js,jsx}\" \"./test/**/*.js\"",
+    "compile": "npm run webpack -- --mode production",
+    "prewebpack": "rimraf ./lib",
+    "webpack": "webpack-compass",
+    "start": "npm run webpack serve -- --mode development",
+    "analyze": "npm run webpack -- --mode production --analyze",
+    "typecheck": "tsc --noEmit",
+    "eslint": "eslint",
+    "prettier": "prettier",
+    "lint": "npm run eslint . && npm run prettier -- --check .",
```
Comment (gribnoysup) [inline packages/compass-import-export/package.json:42]: This command is currently failing in CI because all the source of the plugin is not matching what prettier is expecting. For `compass-aggregations` I deliberately decided not to add prettier to avoid introducing changes to all the source code in the package, it's also fine if we want to add it here, but you would need to run `reformat` command to fix these issues

File: packages/compass/package.json [hunk 9]
```diff
@@ -173,7 +173,7 @@
     "@mongodb-js/compass-field-store": "^7.20.0",
     "@mongodb-js/compass-find-in-page": "^3.20.0",
     "@mongodb-js/compass-home": "^5.23.0",
-    "@mongodb-js/compass-import-export": "^6.21.0",
+    "@mongodb-js/compass-import-export": "^0.1.0",
```
Comment (gribnoysup) [inline packages/compass/package.json:176]: This should stay aligned with the package version

File: packages/compass-import-export/package.json (hunk 8 above)
Comment (leorossi) [reply packages/compass-import-export/package.json:42]: I chose to run `prettier` and commit all changes. This will blow up number of files in this PR but we'll keep formatting consistent with the other packages.

**Comment by leorossi** [comment]
Closing this in favour of #2908
